enumerables.rb

```diff
@@ -0,0 +1,152 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+# rubocop:disable Lint/AmbiguousBlockAssociation
+
+module Enumerable
+  def my_each
+    block_given? ? size.times { |i| yield(to_a[i]) } : (return to_enum)
+    self
+  end
+
+  def my_each_with_index
+    block_given? ? size.times { |i| yield(to_a[i], i) } : (return to_enum)
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    array = []
+    my_each { |value| array << value if yield value }
+    array
+  end
+
+  def my_all?(arg = nil)
+    if block_given?
+      my_each { |value| return false unless yield value }
+    elsif arg.is_a? Regexp
+      my_each { |value| return false unless value.to_s =~ arg }
+    elsif arg.is_a? Class
+      my_each { |value| return false unless value.is_a? arg }
+    elsif arg
+      my_each { |value| return false unless value == arg }
+    elsif arg.nil?
+      my_each { |value| return false unless value }
+    else
+      my_each { |value| return false unless value }
+    end
+    true
+  end
+
+  def my_any?(args = nil)
+    if block_given?
+      my_each { |element| return true if yield(element) }
+      false
+    elsif arg.nil?
+      my_each { |element| return true if n.nil? || element == true }
+    elsif !arg.nil? && (args.is_a? Class)
+      my_each { |element| return true if element.instance_of?(args) }
+    elsif !arg.nil? && args.instance_of?(Regexp)
+      my_each { |element| return true if arg.match(element) }
+    else
+      my_each { |element| return true if element == args }
+    end
+    false
+  end
```

Comment:
- [ ] `my_any` is not working as the original `any?` method, also is not passing all test cases. 

### `any?` is an Enumerable that has these characteristics:
- Returns `true` if at least one element is not false.
```
[1, false, nil].any?
# true
```
- Returns `false` if all elements are false.
```
[nil, false, nil].any?
# false
```
- it doesn't change the original array/range/hash
- it can accept as arguments a Class, a Regex, or a pattern, and returns true or false depending on if the values in the array/range/hash match the condition
- it can use a block, but it is not necessary to make the method work

[Enumerable#any?](https://ruby-doc.org/core-2.7.2/Enumerable.html#method-i-any-3F)